src/mkfs.fat.c

```diff
@@ -1170,17 +1170,26 @@ static void setup_tables(void)
 	if (de->name[0] == 0xe5)
 	    de->name[0] = 0x05;
 	de->attr = ATTR_VOLUME;
-	if (!invariant)
+	if (create_time != (time_t)-1) {
```

Comment:
create_time is initialized to -1 (and is of type time_t), so what is the purpose of this check? It will never be true, right?

Reply:
See: https://github.com/dosfstools/dosfstools/pull/94/files#diff-47b7a74c103277bcad9443e559f1ee9bR1375

`create_time` is initialized to `-1`. And later is updated to up-to-date value, but only in case system is not affected by Year 2038 bug.